File: Recursion/palindrome.js

//use recursion to determine whether a string is a palindrome

var isPal = function(str) {

	var isChar = function(str) {
		var str = str.toLowerCase();
		result ="";
		var letters = /^[a-z]$/
		for (var i = 0; i < str.length; i++) {
			if (letters.test(str[i])) {
				result += str[i];
			}
		}
// 		console.log(result);
		return result;
	}
	
	var pal = function(str) {
	   // console.log(str);
	    if (str.length <= 1) {
	        return true;
	    }
	    else if (str[0] === str[str.length-1]) {
	        return pal(str.slice(1,str.length-1));
	    } else {
	        return false;
	    }
	}
	return pal(isChar(str));
}


var str = "Able was I ere I saw Elba";
console.log(isPal(str)); //true
console.log(isPal("ABCCBA")); //true
console.log(isPal('I\'m not a palindrome')); //false

File: Python/Recursion/Palindrome.py
#use recursion to det. whether a string is a palindrome

def isPalindrome(str):
	def toChars(str):
		s = s.lower()
		ans = ''
		for char in str:
			if char in 'abcdefghijklmnopqrstuvwxyz':
				ans = ans + c
		return ans

def isPal(str):
	if len(str) <= 1:
		return True
	else:
		return str[0] == str[-1] and isPal(str[1:-1])

return isPal(toChars(str))